Firestore/core/src/firebase/firestore/util/status.h

```diff
@@ -93,6 +93,25 @@ class ABSL_MUST_USE_RESULT Status {
   // the floor.
   void IgnoreError() const;
 
+  // Returns a Status that is identical to 's' except that the error_message()
+  // has been augmented by adding 'msg' to the end of the original error
+  // message.
+  //
+  // Annotate should be used to add higher-level information to a Status.  E.g.,
+  //
+  //   util::Status s = file::GetContents(...);
+  //   if (!s.ok()) {
+  //     return s.Annotate("loading blacklist");
+  //   }
+  //
+  // Annotate() adds the appropriate separators, so callers should not include a
+  // separator in 'msg'. The exact formatting is subject to change, so you
+  // should not depend on it in your tests.
+  //
+  // OK status values have no error message and therefore if 's' is OK, the
+  // result is unchanged.
+  Status Annotate(absl::string_view msg) const;
```

Comment:
Note that Status objects aren't immutable. (Though glancing at it, the only non-const method is Update.) As such, this may be better written as `void Annotate(string_view msg); //non-const`.

Or alternatively, we could make Status immutable and alter Update instead, though that seems to diverge a bit from the other existing status implementations (if we care).

Looking at the g3 implementation, it defines this as a free function (and therefore takes the existing status as a parameter). So that's another option too. (Almost equivalently, you could also define it as a static function.)

Basically, I'm not wild about the fact that Update and Annotate treat this as mutable/immutable; they should behave similarly. Any of the above would resolve that for me.

Reply:
Agreed. I've backed this out. `CausedBy` will treat Status as mutable (to match `Update`).